tools/generate_validation_tolerances.py

```diff
@@ -0,0 +1,64 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024-present NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""
+This script reproduces the computation used for generating the validation tolerances in csrc/validator_utils.h:
+https://github.com/NVIDIA/Fuser/blob/892b7ac5646b3b873f13f2b2ea1db15fa9b9effb/csrc/validator_utils.h#L25-L46
+
+The computation (randn + mul + sum) is repeated a large number of times.
+The base tolerance is the max error seen when comparing the desired precision type with the next higher precision.
+The final tolerances are computed by adding a safety factor of 3/4 to this base tolerance.
```

Comment:
```suggestion
```
I believe this talks about https://github.com/NVIDIA/Fuser/blob/d5b2a245cf5a5b3d66b931f9054f14cad3416bdc/csrc/validator_utils.cpp#L23 and is probably outdated. Even if it's up to date, it can easily get outdated because this comment is too far from validator_utils.cpp and nobody will remember to update here when code changes. Therefore, I recommended we remove this comment or make it vague (and less likely to get outdated) by saying the final tolerances will be computed by getTolerances based on this.

Reply:
No, this is not what is being referred to. This script is solely to generate the constants listed in `csrc/validator_utils.h`: https://github.com/NVIDIA/Fuser/blob/892b7ac5646b3b873f13f2b2ea1db15fa9b9effb/csrc/validator_utils.h#L25-L46. This is unrelated to `getTolerance` logic.

If the max error seen is `x`, then the constant listed in the file are `safety_factor * x`.
`getTolerances` does an analysis to figure out what is the reduction size.

Reply:
Who does `safety_factor * x`? I failed to find it in this script. Yet another script?

Reply:
I can include it in this script as well. The reason for keeping it separate was, because, I don't actually know the safety factor used at the time of generating these numbers (it is not documented). When I reverse-engineered it, it would fall between 3-5.

I was using this script as follows:
- Generate the base tolerances
- Interactively experiment with different safety factors and run the tests to see how they impact validation.

Note that this script is a reproduction based on the comments to generate these validation constants for experimentation (I believe we do not have the actual script that was used). The numbers in this file have not been changed.

Reply:
Got it -- thanks for clarification. 

I'm all for checking in everything needed to reproduce the tolerances. That's the only way to avoid tribal knowledge. I understood that the **current** tolerances in `validator_utils.h` are no longer reproducible -- we lost the script. But let's try our best to avoid that for the **future** ones. 

For this PR, if you think a safety factor will be necessary to generate the tolerances to be directly used by validator_utils.h, code that up in the script and check it in. Even if you are not sure what's the right value, put your best guess and iterate.